workflow/common/common.go

```diff
@@ -132,6 +132,8 @@ const (
 	GlobalVarWorkflowParameters = "workflow.parameters"
 	// LocalVarPodName is a step level variable that references the name of the pod
 	LocalVarPodName = "pod.name"
+	// LocalVarPodRetryAttempt is a step level variable that references the retry attempt number if retryStrategy is specified
+	LocalVarPodRetryAttempt = "retryAttempt"
```

Comment:
I feel like we need some sort of name spacing here equivalent to `pod` like `local.retryAttempt` to avoid collisions with user provided variable names?

Reply:
Like I mentioned, our convention for "local" is simply to omit any prefixes. Other examples include `{{item}}`, `{{duration}}`, `{{status}}`.

Not sure what you mean by "user defined variables". The `{{...}}` syntax signifies an Argo variable. If you mean parameter/artifact names, they are always preceded by something like `{{inputs.artifacts.<NAME>...}}`

Also, please rename the variable:
```suggestion
	LocalVarRetryAttempt = "retryAttempt"
```

Reply:
fixed.